examples/matching-engine/tests/basic_operations.rs

```diff
@@ -0,0 +1,371 @@
+        owner_a.to_value()
+    );
+    let response = matching_chain.graphql_query(matching_id, query).await;
+    let orders = response.response["accountInfo"]["entry"]["value"]["orders"]
+        .as_array()
+        .expect("Should have orders array");
+    assert_eq!(orders.len(), 1);
+    let order_id = orders[0].as_u64().unwrap();
+
+    // Cancel the order
+    let cancel_order = Order::Cancel {
+        owner: owner_a,
+        order_id,
+    };
+    let operation = Operation::ExecuteOrder {
+        order: cancel_order,
+    };
+    let cert2 = user_chain_a
+        .add_block(|block| {
+            block.with_operation(matching_id, operation);
+        })
+        .await;
+
+    matching_chain
+        .add_block(|block| {
+            block.with_messages_from(&cert2);
+        })
+        .await;
+
+    user_chain_a.handle_received_messages().await;
+
+    // The order should be cancelled successfully
+}
+
+#[tokio::test]
+async fn test_modify_order() {
+    let (
+        _validator,
+        matching_id,
+        _token_id_a,
+        _token_id_b,
+        user_chain_a,
+        _user_chain_b,
+        matching_chain,
+    ) = setup_matching_engine().await;
+
+    let owner_a = AccountOwner::from(user_chain_a.public_key());
+
+    // Insert an order
+    let order = Order::Insert {
+        owner: owner_a,
+        amount: Amount::from_tokens(20),
+        nature: OrderNature::Bid,
+        price: Price { price: 100 },
+    };
+    let operation = Operation::ExecuteOrder { order };
+    let cert1 = user_chain_a
+        .add_block(|block| {
+            block.with_operation(matching_id, operation);
+        })
+        .await;
+
+    matching_chain
+        .add_block(|block| {
+            block.with_messages_from(&cert1);
+        })
+        .await;
+
+    user_chain_a.handle_received_messages().await;
+
+    // Get the order ID
+    let query = format!(
+        "query {{ accountInfo {{ entry(key: {}) {{ value {{ orders }} }} }} }}",
+        owner_a.to_value()
+    );
+    let response = matching_chain.graphql_query(matching_id, query).await;
+    let orders = response.response["accountInfo"]["entry"]["value"]["orders"]
+        .as_array()
+        .unwrap();
+    let order_id = orders[0].as_u64().unwrap();
+
+    // Modify the order (reduce quantity)
+    let modify_order = Order::Modify {
+        owner: owner_a,
+        order_id,
+        cancel_amount: Amount::from_tokens(5),
+    };
+    let operation = Operation::ExecuteOrder {
+        order: modify_order,
+    };
+    let cert2 = user_chain_a
+        .add_block(|block| {
+            block.with_operation(matching_id, operation);
+        })
+        .await;
+
+    matching_chain
+        .add_block(|block| {
+            block.with_messages_from(&cert2);
+        })
+        .await;
+
+    user_chain_a.handle_received_messages().await;
+
+    // The order should be modified successfully
```

Comment:
Shouldn't we check at this point that five tokens have been refunded and the order now has an amount of 95?